catalog/video/catalog-mpegts-stream/src/main/java/com/connexta/alliance/video/stream/mpegts/DateTemplateFilenameGenerator.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (c) Connexta, LLC
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.alliance.video.stream.mpegts;
+
+import static org.apache.commons.lang3.Validate.notNull;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Replaces one or more tokens within a filename with the current time. The format of the token is
+ * <code>%{date=FMT}</code>, where <code>FMT</code> is format string defined by
+ * {@link SimpleDateFormat}.
+ */
+public class DateTemplateFilenameGenerator extends BaseFilenameGenerator {
+
+    private static final Logger LOGGER =
+            LoggerFactory.getLogger(DateTemplateFilenameGenerator.class);
+
+    private static final String PREFIX = "%{date=";
+
+    private static final String SUFFIX = "}";
+
+    private static final String REGEX = ".*" + Pattern.quote(PREFIX) + "([^}]+)" + Pattern.quote(
+            SUFFIX) + ".*";
+
+    /**
+     * This capture group number must agree with the groupings in {@link #REGEX}.
+     */
+    private static final int CAPTURE_GROUP = 1;
+
+    /**
+     * By default, new Date objects are created by calling {@link Date#Date()}. This can
+     * be changed by setting a new {@link DateSupplier} with {@link #setDateSupplier(DateSupplier)}.
+     */
+    private DateSupplier dateSupplier = Date::new;
+
+    /**
+     * Set an alternative DateSupplier to the default.
+     *
+     * @param dateSupplier must be non-null
+     */
+    public void setDateSupplier(DateSupplier dateSupplier) {
+        notNull(dateSupplier, "dateSupplier must be non-null");
+        this.dateSupplier = dateSupplier;
+    }
+
+    @Override
+    protected String doGenerateFilename(String baseFilename) {
+
+        String tmp = baseFilename;
+
+        while (tmp.matches(REGEX)) {
+            Pattern p = Pattern.compile(REGEX);
+            Matcher m = p.matcher(tmp);
+            if (m.matches()) {
+                String fmt = m.group(CAPTURE_GROUP);
+                SimpleDateFormat sdf = new SimpleDateFormat(fmt);
```

Comment:
You could use [`FastDateFormat`](https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/time/FastDateFormat.html) which is thread-safe. Then you could just make it a static member and not have to create a new one each time.